lib/scotty.js

```diff
@@ -40,73 +42,91 @@ function createOrUpdateBucket(bucket, region, update, force, quiet, logger) {
     })
 }
 
-function scotty(source, bucket, region, website, spa, update, force, quiet, logger) {
-  if (!source || !bucket || !region)
+function scotty(options) {
+  if (!options || !options.source || !options.bucket || !options.region)
     return Promise.reject('Source, bucket and region are required')
 
-  return createOrUpdateBucket(bucket, region, update, force, quiet, logger)
+  return createOrUpdateBucket(options)
     .then(result => {
-      if (!quiet)
-        logger.log('   bucket'.magenta, '✤', colors.cyan(result))
+      if (!options.quiet)
+        options.logger.log('   bucket'.magenta, '✤', colors.cyan(result))
 
       time = new Date().getTime()
-      return upload(source, bucket)
+      return upload(options.source, options.bucket)
     })
     .then(() => {
       return new Promise((resolve, reject) => {
-        return getFolderSize(source, (err, size) => {
+        return getFolderSize(options.source, (err, size) => {
           if (err)
             return reject(err)
 
           const duration = ((new Date().getTime() - time) / 1000).toFixed(2)
 
-          if (!quiet)
-            logger.log('   upload'.magenta, '✤', colors.cyan('completed'), `(${getFormattedSize(size)}, ${duration}s)`.cyan.dim)
+          if (!options.quiet)
+            options.logger.log('   upload'.magenta, '✤', colors.cyan('completed'), `(${getFormattedSize(size)}, ${duration}s)`.cyan.dim)
 
           resolve()
         })
       })
     })
     .then(() => {
-      if (website || spa) {
-        if (!quiet)
-          logger.log('   config'.magenta, '✤', 'set as a website'.cyan)
+      if (options.website || options.spa) {
+        if (!options.quiet)
+          options.logger.log('   config'.magenta, '✤', 'set as a website'.cyan)
 
-        return setAsWebsite(bucket, spa)
+        return setAsWebsite(options.bucket, options.spa)
           .then(() => {
-            if (!quiet)
-              logger.log('   config'.magenta, '✤', 'set cdn'.cyan)
+            if (!options.quiet)
+              options.logger.log('   config'.magenta, '✤', 'set cdn'.cyan)
 
-            return setCdn(bucket)
+            return setCdn(options.bucket)
           })
           .then(domainName => {
             return {
               cdn: true,
+              domain: domainName,
               url: 'https://' + domainName
             }
           })
       }
 
       return true
     })
+    .then( response => {
+      return new Promise((resolve) => {
+        if( response.domain && options.zoneId && options.domain ) {
```

Comment:
just minor code styling. there are also some white spaces before `response` and here inside the brackets, instead adding them just before `if`s bracket. to keep everything in single style